src/gui/guiButton.cpp

```diff
@@ -750,22 +753,26 @@ void GUIButton::setFromStyle(const StyleSpec& style)
 	bool pressed = (style.getState() & StyleSpec::STATE_PRESSED) != 0;
 
 	if (style.isNotDefault(StyleSpec::BGCOLOR)) {
-
 		setColor(style.getColor(StyleSpec::BGCOLOR));
 
 		// If we have a propagated hover/press color, we need to automatically
 		// lighten/darken it
 		if (!Styles[style.getState()].isNotDefault(StyleSpec::BGCOLOR)) {
-			for (size_t i = 0; i < 4; i++) {
 				if (pressed) {
-					Colors[i] = multiplyColorValue(Colors[i], COLOR_PRESSED_MOD);
+					BgColor = multiplyColorValue(BgColor, COLOR_PRESSED_MOD);
+
+					for (size_t i = 0; i < 4; i++)
+						Colors[i] = multiplyColorValue(Colors[i], COLOR_PRESSED_MOD);
 				} else if (hovered) {
-					Colors[i] = multiplyColorValue(Colors[i], COLOR_HOVERED_MOD);
+					BgColor = multiplyColorValue(BgColor, COLOR_HOVERED_MOD);
+
+					for (size_t i = 0; i < 4; i++)
+						Colors[i] = multiplyColorValue(Colors[i], COLOR_HOVERED_MOD);
 				}
-			}
 		}
 
 	} else {
+		BgColor = video::SColor(255, 255, 255, 255);
 		for (size_t i = 0; i < 4; i++) {
 			video::SColor base =
 					Environment->getSkin()->getColor((gui::EGUI_DEFAULT_COLOR)i);

```

Comment:
`BgColor` should be set to what `base` is set to instead of white.  It doesn't make a difference now, but if `BgColor` is used in the future somewhere else, this could cause a strange bug.